web/Controllers/PropertyController.cs

```diff
@@ -36,22 +36,26 @@ public ActionResult DeleteProperty(int propertyId)
         var propDB = _propertyService.GetPropertyByPropertyId(propertyId);
         if (propDB == null)
         {
-            _logger.LogWarning("Property with ID {PropertyId} does not exist", propertyId);
-            return StatusCode(404);
+            _logger.LogWarning($"Property with Id {propertyId} does not exist");
+            AddFlashMessage("danger", $"Property with Id {propertyId} does not exist");
+            if (GetCurrentUser().LandlordId != null)
+            {
+                return RedirectToAction("ViewProperties", "Landlord", new { id = GetCurrentUser().LandlordId });
+            }
+            return RedirectToAction("Index", "Home");
         }
 
         if (GetCurrentUser().IsAdmin == false && GetCurrentUser().LandlordId != propDB.LandlordId)
         {
-            _logger.LogWarning("You do not have access to any property with ID {PropertyId}.", propertyId);
-            return StatusCode(404);
+            _logger.LogWarning($"User {GetCurrentUser().Id} does not have access to any property with ID {propertyId}.");
+            return StatusCode(405);

```

Comment:
I think instead of a 405 this should be a 403, as the current user is not allowed to access this (ie Forbidden) not that the server knows the request method, but the target resource doesn't support this method (ie Method Not Allowed) as that would be saying that the property would be unable to delete itself, not that the landlord was unauthorised in deleting it

Reply:
Yes, that's just me mis-remembering the correct code - I meant 403.